### PR TITLE
Enable wayland by default, fallback to X11

### PR DESCRIPTION
**Pull request by ghisvail**

### CODE REVIEW

File: com.bitwarden.desktop.yaml [hunk 1]
```diff
@@ -51,8 +51,12 @@ modules:
...
         dest-filename: bitwarden.sh
         commands:
-          - export TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"
-          - exec zypak-wrapper /app/Bitwarden/bitwarden "$@"
+          - export TMPDIR="${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}"
+          - if [ "${XDG_SESSION_TYPE}" == "wayland" ]; then
+          - zypak-wrapper /app/Bitwarden/bitwarden --enable-features=UseOzonePlatform --ozone-platform=wayland "$@"
+          - else
+          - zypak-wrapper /app/Bitwarden/bitwarden "$@"
+          - fi
```
Comment (TheEvilSkeleton) [inline com.bitwarden.desktop.yaml:59]: Just my two cent: the structure of it is displeasing. I suggest writing a separate bash script.

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

**Review by DoodlesEpic — APPROVED** [review]

**Comment by ghisvail** [comment]
@proletarius101 thanks for the heads up :+1:

**Review by alistair23 — APPROVED** [review]

**Comment by ghisvail** [comment]
> is there anything blocking this from getting merged?

Users testing it and confirming it works both on Wayland and X11.

**Comment by taaem** [comment]
@ghisvail I just tested the latest build and there are 2 things that are broken:
1. We need to tell electron where to find the cursors, otherwise the cursor is tiny on wayland. Fix (taken from https://github.com/flathub/org.signal.Signal/blob/master/org.signal.Signal.yaml): 
```yaml
  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
```
2. There is an exception because the app can't access the system bus:
```
[2:0905/134918.197642:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Datei oder Verzeichnis nicht gefunden
```
This can be fixed by allowing the flatpak access to the system bus (see https://github.com/flathub/org.signal.Signal/issues/280):
```yaml
  - --socket=system-bus
```

**Comment by taaem** [comment]
The second one should not be fixed like that but instead like this (see https://github.com/flathub/com.visualstudio.code/blob/master/com.visualstudio.code.yaml#L26C1-L26C46):
```yaml
  - --system-talk-name=org.freedesktop.login1
```

**Comment by taaem** [comment]
Sorry for the update again, but according to https://github.com/electron/electron/blob/main/shell/browser/lib/power_observer_linux.cc#L18 the correct fix would be:
```yaml
  - --system-talk-name=org.freedesktop.login1
  - --system-talk-name=org.freedesktop.login1.Manager
```

**Comment by ghisvail** [comment]
> Sorry for the update again, but according to https://github.com/electron/electron/blob/main/shell/browser/lib/power_observer_linux.cc#L18 the correct fix would be:
> 
> ```yaml
>   - --system-talk-name=org.freedesktop.login1
>   - --system-talk-name=org.freedesktop.login1.Manager
> ```

Thank you for investigating this. Could you submit this fix in a separate PR?

**Comment by taaem** [comment]
@ghisvail I will in a second :)

**Comment by taaem** [comment]
@ghisvail should I also add the other fix in the new pr?

**Comment by ghisvail** [comment]
> 

yes please

**Comment by ghisvail** [comment]
@taaem could you test it again, now that your changes are in?

**Comment by taaem** [comment]
@ghisvail everything works for me now on wayland :)

**Comment by cassidyjames** [comment]
Testing locally on Endless OS on Wayland, and it works. It seems to fix #165 for me as well.

**Comment by ghisvail** [comment]
Looks like it's working on my X11 session too.

**Pull request merged**